e2e/nomostest/config-sync.go

```diff
@@ -1255,96 +1257,147 @@ func resetRepository(nt *NT, repoType RepoType, nn types.NamespacedName, upstrea
+	defaultRS.Spec.DeepCopyInto(&rs.Spec)
+	if err = nt.Update(rs); err != nil {
+		nt.T.Fatal(err)
+	}
+	// nt.MustMergePatch(rs, fmt.Sprintf(`{"spec": {"sourceFormat": "%s", "git": {"dir": "%s"}}}`, sourceFormat, AcmeDir))
+
+	// Resume the reconciler-manager, which will update and scale up the reconciler.
+	scaleDeployment(nt, reconcilermanager.ManagerName, configmanagement.ControllerNamespace, 1)
 }
 
-// resetRootRepos resets all RootSync objects to the initial commit to delete
-// managed testing resources.
-//
-// The spec.sourceFormat is reset to the specified format, to ensure the
-// remaining objects are deleted without erroring from format mismatch.
-//
-// The spec.git.dir is reset to "acme", which is created by
-// NewRepository and Repository.ReInit and must still exist.
-//
-// TODO: reset everything else, including the branch
-func resetRootRepos(nt *NT, upstream string, sourceFormat filesystem.SourceFormat) {
-	// Reset the default root repo first so that other managed RootSyncs can be re-created and initialized.
+func resetNamespaceRepo(nt *NT, rsNN types.NamespacedName, upstream string, sourceFormat filesystem.SourceFormat) {
+	rs := &v1beta1.RepoSync{}
+	err := nt.Get(rsNN.Name, rsNN.Namespace, rs)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			nt.T.Fatal(err)
+		}
+		// If not found, reset the Repository and create the RepoSync.
+		// This can happen if it was managed by another RepoSync which has
+		// already been reset.
+		nt.NonRootRepos[rsNN] = resetRepository(nt, NamespaceRepo, rsNN, upstream, sourceFormat)
+		defaultRS := RepoSyncObjectV1Beta1FromNonRootRepo(nt, rsNN)
+		if err = nt.Create(defaultRS); err != nil {
+			nt.T.Fatal(err)
+		}
+		return
+	}
+
+	defaultRS := RepoSyncObjectV1Beta1FromNonRootRepo(nt, rsNN)
+	if cmp.Equal(rs.Spec, defaultRS.Spec) {
+		// If found and unchanged, just reset the Repository.
+		// This can happen if the RepoSync was not modified by the test.
+		nt.NonRootRepos[rsNN] = resetRepository(nt, NamespaceRepo, rsNN, upstream, sourceFormat)
+		return
+	}
+
+	// If found and needs changes, we need to pause reconciliation to avoid a
+	// race condition between the reconciler-manager and the reconciler.
+
+	// Pause the reconciler-manager.
+	// This allows us to reset the RepoSync Git config without updating the reconciler.
+	scaleDeployment(nt, reconcilermanager.ManagerName, configmanagement.ControllerNamespace, 0)
+	// Pause the reconciler.
+	// This allows us to reset the Git repo without syncing.
+	scaleDeployment(nt, core.NsReconcilerName(rsNN.Namespace, rsNN.Name), configmanagement.ControllerNamespace, 0)
+	nt.NonRootRepos[rsNN] = resetRepository(nt, NamespaceRepo, rsNN, upstream, sourceFormat)
+
+	// Reset the RepoSync spec to default
+	defaultRS.Spec.DeepCopyInto(&rs.Spec)
+	if err = nt.Update(rs); err != nil {
+		nt.T.Fatal(err)
+	}
+	// nt.MustMergePatch(rs, fmt.Sprintf(`{"spec": {"sourceFormat": "%s", "git": {"dir": "%s"}}}`, sourceFormat, AcmeDir))
```

Comment:
remove

Reply:
Yup. Left it in to see if you were paying attention. ;)